galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistQuery.java

```diff
@@ -0,0 +1,173 @@
+package dev.galasa.framework.api.webui;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property =
+{"osgi.http.whiteboard.servlet.pattern=/webui/worklist" }, name = "Galasa Worklist microservice")
+public class WorklistQuery extends HttpServlet {
+
+	private static final long serialVersionUID = 1L;
+	  
+	@Reference
+	IFramework framework;
+	
+	final static Gson gson = GalasaGsonBuilder.build();
+    
+    private WorklistUtils worklistUtils;
+
+	
+	@Override 
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		
+		String principalUsername = worklistUtils.getPrincipalUsername(req);
+		
+		returnWorklist(principalUsername, resp, 200);
+
+	}
+	
+	
+	@Override
+	protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		
+		String principalUsername = worklistUtils.getPrincipalUsername(req);
+		
+		Map<String, String> paramMap = getParameterMap(req);
+	
+		String runId = "";
+		if (paramMap.get("runId") != null && !paramMap.get("runId").equals("")) {
+			runId = paramMap.get("runId");
+		}
+		
+
+		try {
+			worklistUtils.addRunIdToDss(principalUsername, runId);
+		} catch (Exception e) {
+			throw new ServletException("Error while updating the DSS", e);
+		}
+		
+		int statusCode = 200;
+		if (worklistUtils.newResourceCreated == true) {
```

Comment:
should return a boolean from the addRunIdToDss instead of accessing the object fields.   But the worklistutilts does not need to exist anyway